rescript/replace_taxonomy.py

```diff
@@ -0,0 +1,50 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from qiime2.plugin import MetadataColumn, Categorical
+import pandas as pd
+import re
+
+
+def make_substitutions_dict(taxonomy_replacement_map):
```

Comment:
IMO it looks like this function can probably be removed... the `to_series()` method seems unnecessary now that you use a metadata column as input (it will already be a series), so this function just transforms the series to a dict, which could probably happen in-line below.

Reply:
t appears I need to use `to_series()` and `to_dict()`, as outlined in the [MetadataColumn docs](https://dev.qiime2.org/latest/api-reference/metadata/#qiime2.metadata.MetadataColumn) for this to work. I assume I am missing something subtle here?

Reply:
yeah you are right, the metadata column does need to be converted to series. to_dict() might not be needed depending on whether you use pd.Series.str.replace or re.sub as discussed below.